### PR TITLE
Adding checklists for the issue tracker

### PR DESCRIPTION
**Pull request by fziegner**
This PR addresses #174 by adding templates for the issue tracker. I checked how projects like pandas and numpy manage this which mostly is in the form of interactive forms using YAML. These don't allow editing the issue while creating it so I opted to use Markdown in the style of the pull request template.

I created a repository [here](https://github.com/fziegner/tobac_checklists/issues) so you can check out how the templates look like in action. Is this what you had in mind @freemansw1? Unfortunately, Markdown doesn't support variables so you'd still have to manually replace the version number in the release template.

* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [ ] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [ ] Have you added NumPy docstrings for newly added functions? 
* [ ] Have you formatted your code using black? 
* [ ] If you have introduced a new functionality, have you added adequate unit tests?
* [ ] Have all tests passed in your local clone? 
* [ ] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

### CODE REVIEW

**Review by freemansw1 — APPROVED** [review]
Thanks for taking this on, @fziegner . This is indeed what I had in mind. I've made a small comment, but I think this is overall ready to be moved out of draft.

File: .github/ISSUE_TEMPLATE/bug-report.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+name: Bug Report
+about: Report a bug in the tobac library
+title: ''
+labels: bug
+assignees: ''
+
+---
+
+* [ ] Have you searched the issue tracker for the same problem?
+* [ ] Have you checked if you're using the latest version?
```
[the remark this answers is not in the document]
Comment (fziegner) [reply .github/ISSUE_TEMPLATE/bug-report.md]: Good suggestion, I added a check.

**Review by w-k-jones — APPROVED** [review]
Looks good, thanks for adding this! We can always add more templates in future (e.g. a feature request template), but I think it's best to add these straight away

**Comment by freemansw1** [comment]
The notebook CI is failing, but it was also doing so before this PR. I've opened a new issue for that (#370). @fziegner are you happy for us to go ahead and merge this?

**Comment by w-k-jones** [comment]
I'm happy to merge as well, as the Jupyter CI issue has nothing to do with this PR

**Comment by fziegner** [comment]
Yes @freemansw1, no objections from me.

**Pull request merged**